lightly/api/api_workflow_datasources.py

```diff
@@ -51,44 +66,60 @@ def _download_raw_files(
                 cursor=cursor,
                 use_redirected_read_url=use_redirected_read_url,
                 **relevant_filenames_kwargs,
-                **kwargs
+                **kwargs,
             )
             cursor = response.cursor
             samples.extend(response.data)
             if progress_bar is not None:
                 progress_bar.update(len(response.data))
-        samples = [(s.file_name, s.read_url) for s in samples]
+        sample_map = {}
+        for s in samples:
+            if re.match("^\.*/", s.file_name):
+                raise ValueError(
+                    f"Absolute file paths like {s.file_name} are not supported"
+                    " in relevant filenames file {relevant_filenames_file_name}"
+                )
+            if s.file_name in sample_map:
+                warnings.warn(
+                    UserWarning(
+                        f"Duplicate filename {s.file_name} in relevant"
+                        " filenames file {relevant_filenames_file_name}"
+                    )
+                )
+            else:
+                sample_map[s.file_name] = s.read_url
+        samples = [(file_name, read_url) for file_name, read_url in sample_map.items()]
```

Comment:
the order should best be the order returned by the API

Reply:
Here's something I didn't know: since version 3.6 Python standard dictionaries have a funky feature.

They maintain insertion order! E.g.

```
>>> a={}
>>> a[5]=9
>>> a[9]=9
>>> a[3]=9
>>> a[2]=9
>>> a[8]=9
>>> a
{5: 9, 9: 9, 3: 9, 2: 9, 8: 9}
```

So, the check for duplicates (and absolute paths, etc) does not change the sample order, and the order returned by the API is maintained.

Dropping the sorting.

Reply:
Dropped.